Firestore/Source/Util/FSTAsyncQueryListener.mm

```diff
@@ -34,10 +35,20 @@ - (instancetype)initWithDispatchQueue:(FSTDispatchQueue *)dispatchQueue
 }
 
 - (FSTViewSnapshotHandler)asyncSnapshotHandler {
+  FSTWeakify(self);
   return ^(FSTViewSnapshot *_Nullable snapshot, NSError *_Nullable error) {
-    [_dispatchQueue dispatchAsync:^{
-      if (!_muted) {
-        _snapshotHandler(snapshot, error);
+    FSTStrongify(self);
+    if (!self) {
+      return;
+    }
+    FSTWeakify(self);
```

Comment:
Is the strongify/weakify dance necessary here (lines 40-44)?

Reply:
This is a specific case where we _want_ a strong reference because we're intending that the returned callback will keep the snapshot listener alive in the `FSTEventManager` whether or not the user releases the `FSTAsyncQueryListener`.

Please
  * Remove the FSTStrongify/FSTWeakify dance (including early returns)
  * Leave the `self->` to make the capture explicit (as the new warning demands)
  * Add a comment like this one so @jdimond and I aren't the only ones who keep this in our heads. Something like:

```
// Retain `self` strongly in resulting snapshot handler so that even if the user releases
// the `FSTAsyncQueryListener` we'll continue to deliver events. This is done specifically
// to facilitate the common case where users just want to turn on notifications "forever"
// and don't want to have to keep track of our handle to keep them going.
```

Absent that concern, you could have omitted the second `FSTWeakify`.

For background: the weak/strong dance around callbacks is required in cases where the callback is being retained but you don't want the caller to necessarily extend the lifetime of the items referenced from the callback.

The weakify/strongify magic macros help with this by avoiding bugs related to keeping two names for the same thing. Without it you have

```objc
__weak typeof(self) weakSelf = self;
dispatch_async(queue, ^{
  typeof(self) strongSelf = self;
  if (!strongSelf) return;

  if (_muted) {} // bug
});
```
The bug is that in addition to the `weakSelf` this also implicitly created a strong reference through the ivar reference. ivars bind to `self` and the only `self` is one in the outer scope.

`FSTStrongify` helps by redeclaring a shadowed `self` in the inner block such that usage of `self` in the block (even via ivar) is bound to that one rather than the self in the outer scope.

However, `FSTStrongify` is just a way of disabling warnings in an assignment like this:

```objc
FSTAsyncQueryListener *self = fstWeakPointerToself;
```

Therefore you could repeat additional shadowing declarations so long as `fstWeakPointerToself` (as declared by `FSTWeakify`) was in scope.

Reply:
Done.